cfme/automate/explorer/domain.py

```diff
@@ -264,6 +278,22 @@ def update(self, updates):
             view.flash.assert_message(
                 'Edit of Automate Domain "{}" was cancelled by the user'.format(self.name))
 
+    def refresh(self, branch_or_tag=None, git_branch=None, cancel=False):
+        view = navigate_to(self, 'Refresh')
+
+        # It refreshes the domain for default values of 'branch_or_tag' and 'git_branch'; if these
+        # values are not provided while calling 'refresh'.
+        select_branch_or_tag = branch_or_tag or view.branch_or_tag.selected_option
```

Comment:
I don't think this is necessary logic. The behavior that you are trying to replicate here is the default behavior of the `View.fill` method.

You can simply fill the DomainRefreshView object that you have from navigating to `Refresh` destination.

```
changed = view.fill(
    {
        'branch_or_tag': branch_or_tag,
        'git_branches': git_branch
    }
)

if changed and not cancel:
    view.save_button.click()
else:
    view.cancel_button.click()
```

`View.fill` will skip taking any action on the widgets when the value to fill is `None`. Since the function takes default values of `None`, when the user doesn't pass them, the currently selected option (default) on those widgets will be left alone.

Reply:
@mshriver 
- Imported Domain refresh must work without providing ```branch_or_tag``` and  ```git_branches``` because after importing this domain. 
- Refresh view dynamically fetches the branches for that domain.
- Hence updating as below:
```
    def refresh(self, branch_or_tag=None, git_branch=None, cancel=False):
        view = navigate_to(self, 'Refresh')

        # It refreshes the domain for default values of 'branch_or_tag' and 'git_branch'; if these
        # values are not provided while calling 'refresh' then imported domain should be refreshed
        # with default values of 'branch_or_tag' and 'git_branches'.

        changed = view.fill(
            {
                'branch_or_tag': branch_or_tag or view.branch_or_tag.selected_option,
                'git_branches': git_branch or view.git_branches.selected_option
            }
        )

        if changed and not cancel:
            view.save_button.click()
        else:
            view.cancel_button.click()
        view.flash.assert_no_error()
```